frontend/src/views/main/project/modals/CreateSliceModal.vue

```diff
@@ -0,0 +1,148 @@
+<template>
+    <vue-final-modal
+        v-slot="{ close }"
+        v-bind="$attrs"
+        classes="modal-container"
+        content-class="modal-content"
+        v-on="$listeners"
+    >
+        <div class="text-center">
+            <v-card>
+                <v-card-title>
+                    Create new slice for "{{ dataset.name ?? dataset.id }}"
+                </v-card-title>
+                <v-card-text>
+                    <ColumnFilterCreator v-for="(columnFilter, idx) in columnFilters"
+                                         :key="idx"
+                                         :dataset="dataset"
+                                         :column-name.sync="columnFilter.column"
+                                         :column-type.sync="columnFilter.columnType"
+                                         :slicing-type.sync="columnFilter.slicingType"
+                                         :value.sync="columnFilter.value"
+                    />
+                </v-card-text>
+                <v-card-actions>
+                    <div class="flex-grow-1"/>
+                    <v-btn color="secondary" @click="close">Cancel</v-btn>
+                    <v-btn color=primary @click="save(close)">Save</v-btn>
+                </v-card-actions>
+            </v-card>
+        </div>
+    </vue-final-modal>
+</template>
+
+<script setup lang="ts">
+
+import {DatasetDTO, SlicingFunctionDTO} from "@/generated-sources";
+import {ref} from "vue";
+import ColumnFilterCreator from "@/views/main/utils/ColumnFilterCreator.vue";
+import {v4 as uuidv4} from 'uuid';
+import {useCatalogStore} from "@/stores/catalog";
+import {NoCodeFilter, NoCodeSlicingType} from "@/utils/no-code-slicing-type.enum";
+
+const props = defineProps<{
+    dataset: DatasetDTO
+}>();
+
+const columnFilters = ref<Array<Partial<NoCodeFilter>>>([{}])
+
+const emit = defineEmits(['created']);
+
+const CLAUSE_BUILDERS: { [clause in NoCodeSlicingType]: (column: string, value: string) => string } = {
+    [NoCodeSlicingType.IS]: (column, value) => `giskard.slicing.slice.EqualTo(${column}, ${value})`,
```

Comment:
I don't think UI should be aware of the python implementation details. Doing it on the backend is better